### PR TITLE
Earn/add user cache

### PR DESCRIPTION
**Pull request by waream2**
this PR adds a caching layer. Specifically it's for when a message is sent, currently, we are hitting the DB for every single message sent in the server. This PR will make it so we cache users and check there first to see if we need to go to the DB or not.

### CODE REVIEW

File: src/events/messageCreate.js [hunk 1]
```diff
@@ -5,8 +5,10 @@ import {
 	reportChannelPoints,
 	givePoint,
 	helpCommand,
+	sendBotIntro,
 } from "../message_replies/index.js";
 import { addUserToPoints } from "../../db.js";
+import { addUserToCache } from '../utils/userCache.js';
 
 
 function matchSufix(str) {
```
Comment (Claudi0-V) [inline src/events/messageCreate.js:14]: change to:
```suggestion
async function matchSufix(str) {
```

File: src/events/messageCreate.js [hunk 2]
```diff
@@ -25,7 +27,12 @@ export default {
 
 		// try to add the user to the points DB, if they are already there
 		// db function will reject this
-		addUserToPoints(interaction.author.id);
+		const isNewUser = addUserToPoints(interaction.author.id);
```
Comment (Claudi0-V) [inline src/events/messageCreate.js:30]:
```suggestion
		const isNewUser = await addUserToPoints(interaction.author.id);
```

File: src/events/messageCreate.js [hunk 3]
```diff
@@ -5,8 +5,10 @@ import {
 	reportChannelPoints,
 	givePoint,
 	helpCommand,
+	sendBotIntro,
 } from "../message_replies/index.js";
 import { addUserToPoints } from "../../db.js";
+import { addUserToCache } from '../utils/userCache.js';
```
Comment (Claudi0-V) [inline src/events/messageCreate.js:11]: I can see the reason for this line (probably legacy):
```suggestion

```

File: src/utils/userCache.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+const checkUserCache = (user) => {
+  if (userCache[user]) {
+    return true;
+  }
+  else {
+    return false;
+  }
+};
```
Comment (Claudi0-V) [inline src/utils/userCache.js:18]: you could replace with
```suggestion
const checkUserCache = (user) => {
  if (userCache[user]) {
    return true;
  }
  return false;
};
```
or
```suggestion
const checkUserCache = (user) => !!userCache[user]);
```

**Review by Claudi0-V — DISMISSED** [review]
that's it for now :)

File: src/utils/userCache.js (hunk 4 above)
Comment (waream2) [reply src/utils/userCache.js:18]: lol i hate implicit `else` statements, makes the code awkward to read IMO, not married to it though. Ill commit the first change, i dont like the `!!`, its easier to read if its verbose. @Claudi0-V

**Review by waream2 — COMMENTED** [review]
@Claudi0-V can you approve it? I can't approve my own one.But it looks good!

**Review by Claudi0-V — APPROVED** [review]

**Pull request merged**